spec/index.md

```diff
@@ -670,37 +680,107 @@ See [The system state tree](#state-tree) for details on the state tree.
--   `method_name` (`text`): Name of the canister method to call
+-   `method_name` (`text`): Name of the canister method to call.
 
--   `arg` (`blob`): Argument to pass to the canister method
+-   `arg` (`blob`): Argument to pass to the canister method.
 
-If the call resulted in a reply, the response is a CBOR (see [CBOR](#cbor)) map with the following fields:
+If the query call resulted in a reply, the response is a CBOR (see [CBOR](#cbor)) map with the following fields:
 
--   `status` (`text`): `replied`
+-   `status` (`text`): `"replied"`
 
 -   `reply`: a CBOR map with the field `arg` (`blob`) which contains the reply data.
 
 If the call resulted in a reject, the response is a CBOR map with the following fields:
 
--   `status` (`text`): `rejected`
+-   `status` (`text`): `"rejected"`
 
 -   `reject_code` (`nat`): The reject code (see [Reject codes](#reject-codes)).
 
 -   `reject_message` (`text`): a textual diagnostic message.
 
--   `error_code` (text): an optional implementation-specific textual error code (see [Error codes](#error-codes)).
+-   `error_code` (`text`): an optional implementation-specific textual error code (see [Error codes](#error-codes)).
+
+### Request: Signed query call {#http-signed-query}
+
+:::note
+
+The signed query call API is considered EXPERIMENTAL. Canister developers must be aware that the API may evolve in a non-backward-compatible way.
+
+:::
+
+A **signed** query call is a query call amended with signatures produced by IC nodes evaluating the query call to certify the query response.
+
+In order to make a signed query call to a canister, the user makes a POST request to `/api/v2/canister/<effective_canister_id>/signed_query`. The request body has the same structure as the request body for [Request: Query call](#http-query) with the `content` map having the following **additional** fields:
+
+-   (optional) `requested_signatures` (`text`): Determines whether and how many signatures are returned in the response; possible values are `"none"`, `"one"`.
+
+-   (optional) `requested_certificate` (`text`): Determines whether a certificate (as specified in [Certification](#certification)) is returned in the response; possible values are `"no"`, `"yes"`.
+
+The response is a CBOR (see [CBOR](#cbor)) with the same fields as for [Request: Query call](#http-query) response and the following **additional** fields:
+
+-   (optional) `signatures` (`[+ node-signature]`): a non-empty list of node signatures for the returned query response.
+
+-   (optional) `certificate` (`blob`): a certificate, as specified in [Certification](#certification).
+
+Unless `requested_signatures` is null or set to `"none"` in the request, the response to a signed query call contains a non-empty list of signatures for the returned response produced by the individual IC nodes that computed the same returned response. Every such signature (whose type is denoted as `node-signature`) is a CBOR (see [CBOR](#cbor)) map with the following fields:
```

Comment:
We can still have the boundary nodes return HTTP error responses, no? I see we haven't covered this for queries or read_state yet. Should we do so?

Reply:
Please see the commit d993538.

Reply:
That makes sense, thanks - but I'm not sure if that's the complete story? For example, what if we try to query a canister that's not in any range in the routing table? I guess we get a 404 back directly from the boundary node? If that's correct, then I'd add something just to be consistent with the update call spec. I don't think we need to resolve this in this PR as you're not introducing the problem, an issue would do.

Reply:
> If that's correct, then I'd add something just to be consistent with the update call spec. I don't think we need to resolve this in this PR as you're not introducing the problem, an issue would do.

Agreed. Here's the issue: https://github.com/dfinity/interface-spec/issues/194

Reply:
This conversation has been resolved offline.